packages/create-cloudflare/src/common.ts

```diff
@@ -325,10 +327,31 @@ export const gitCommit = async (ctx: PagesGeneratorContext) => {
 	if (!(await isGitInstalled()) || !(await isInsideGitRepo(ctx.project.path)))
 		return;
 
+	const git = simpleGit();
+
+	await git.add(".");
+
+	const gitStatus = await simpleGit().status();
+
+	if (gitStatus.not_added.length > 0) {
+		// there are files that haven't been staged when we did `git add .`
+		// meaning that we are in a monorepo provide a warning and don't commit
+		await simpleGit().reset(["."]);
+		logRaw(
+			`${gray(shapes.leftT)} ${
+				status.warning
+			} The Creation of a git commit failed since we're inside a larger git repository, please commit the changes manually\n${gray(
+				shapes.bar
+			)} `
+		);
+		ctx.insideMonorepo = true;
```

Comment:
I think this flag, renamed to `insidePreExistingRepo`, should be set when we discover we're in a git repo and skip creating one ourselves

This is a very side-effecty way of doing this. Better to be clear and concrete